docs/src/prototypes/chatMessageWithPopover/Popover.tsx

```diff
@@ -0,0 +1,183 @@
+  createComponent,
+  ComponentSlotStyle,
+  ComponentVariablesInput,
+  toolbarButtonBehavior,
+} from '@stardust-ui/react'
+import { ReactChildren } from 'types/utils'
+import * as React from 'react'
+import * as cx from 'classnames'
+
+export interface PopoverProps {
+  className?: string
+}
+
+interface PopoverState {
+  focused: boolean
+  popupOpened: boolean
+}
+
+class Popover extends React.Component<PopoverProps, PopoverState> {
+  state = {
+    focused: false,
+    popupOpened: false,
+  }
+
+  changeFocusState = (isFocused: boolean) => {
+    this.state.focused !== isFocused && this.setState({ focused: isFocused })
+  }
+
+  handleFocus = () => {
+    this.changeFocusState(true)
+  }
+
+  handleBlur = e => {
+    // if e.relatedTarget === null, so the click was outside this container
+    if (!this.state.popupOpened || e.relatedTarget === null) {
+      const shouldPreserveFocusState = e.currentTarget.contains(e.relatedTarget)
+      this.changeFocusState(shouldPreserveFocusState)
+    } else {
+      e.stopPropagation()
+    }
+  }
+
+  handleMenuClick = () => {
+    // close popup when other MenuItem clicked, but the event propagation was stopped
+    this.state.popupOpened && this.setState({ popupOpened: false })
+  }
+
+  menuStyles = ({ theme: { siteVariables } }) => ({
+    transition: 'opacity 0.2s',
+    position: 'absolute',
+    top: '-20px',
+    right: '5px',
+    background: siteVariables.white,
+    boxShadow: '0px 2px 4px #ddd',
+    borderRadius: '.3rem',
+    opacity: 0,
+
+    '& .smile-emoji': {
+      display: 'none',
+    },
+
+    '&.focused .smile-emoji': {
+      display: 'flex',
+    },
+
+    '&:hover .smile-emoji': {
+      display: 'flex',
+    },
+  })
+
+  render() {
+    return (
+      <Menu
+        styles={this.menuStyles}
+        iconOnly
+        className={cx(this.props.className, this.state.focused ? 'focused' : '')}
+        items={[
+          {
+            key: 'smile',
+            icon: 'smile',
+            className: 'smile-emoji',
+            accessibility: toolbarButtonBehavior,
+            'aria-label': 'smile one',
+          },
+          {
+            key: 'smile2',
+            icon: 'smile',
+            className: 'smile-emoji',
+            accessibility: toolbarButtonBehavior,
+            'aria-label': 'smile two',
+          },
+          {
+            key: 'smile3',
+            icon: 'smile',
+            className: 'smile-emoji',
+            accessibility: toolbarButtonBehavior,
+            'aria-label': 'smile three',
+          },
+          {
+            key: 'a',
+            icon: 'thumbs up',
+            accessibility: toolbarButtonBehavior,
+            'aria-label': 'thumbs up',
+          },
+          {
+            key: 'c',
+            icon: 'ellipsis horizontal',
+            accessibility: toolbarButtonBehavior,
+            'aria-label': 'more options',
+          },
+        ]}
+        renderItem={this.renderItemOrContextMenu}
```

Comment:
Watch out, we merge the PR for removing the renderXXX methods from the component's API. This should be refactored. For more info, please see the examples on the PR https://github.com/stardust-ui/react/pull/519 and the Issue https://github.com/stardust-ui/react/issues/355

Reply:
Yeah, I just found out it yesterday when Popup hasn't render. Will take a look today how to solve it